File: C#1/Loops/HexToDecimal/HexToDecimal.cs
using System;

class HexToDecimal
{
    static void Main()
    {
        string hex = Console.ReadLine();
        long number = 0;
        for (int i = hex.Length - 1; i >= 0; i--)
        {
            switch (hex[i])
            {
                case '0': number += (long)(0 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '1': number += (long)(1 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '2': number += (long)(2 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '3': number += (long)(3 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '4': number += (long)(4 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '5': number += (long)(5 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '6': number += (long)(6 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '7': number += (long)(7 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '8': number += (long)(8 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case '9': number += (long)(9 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'A': number += (long)(10 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'B': number += (long)(11 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'C': number += (long)(12 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'D': number += (long)(13 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'E': number += (long)(14 * Math.Pow(16, (hex.Length - 1 - i))); break;
                case 'F': number += (long)(15 * Math.Pow(16, (hex.Length - 1 - i))); break;
            }
        }
        Console.WriteLine("{0}", number);
    }
}

File: JavaScript Fundamentals/Loops/HexToDecimal.js
function solve(args){
    var number = String(args[0]);
    var output = 0;
    for(var i = 0, len = number.length; i < len; i++)
    {
        output += parseInt(findNum(number[i]) * Math.pow(16, len - i - 1));
    }
    console.log(output);
}

function findNum(symbol)
{
    switch(symbol)
    {
        case '0': return 0;
        case '1': return 1;
        case '2': return 2;
        case '3': return 3;
        case '4': return 4;
        case '5': return 5;
        case '6': return 6;
        case '7': return 7;
        case '8': return 8;
        case '9': return 9;
        case 'A': return 10;
        case 'B': return 11;
        case 'C': return 12;
        case 'D': return 13;
        case 'E': return 14;
        case 'F': return 15;
        
    }
}

solve(['FE']);